File: src/main/java/de/spries/fleetcommander/model/facade/OtherPlayer.java
package de.spries.fleetcommander.model.facade;

import java.util.List;
import java.util.stream.Collectors;

import de.spries.fleetcommander.model.core.Player;
import de.spries.fleetcommander.model.core.Player.Status;

public class OtherPlayer {

	protected Player originalPlayer;

	public OtherPlayer(Player originalPlayer) {
		this.originalPlayer = originalPlayer;
	}

	public String getName() {
		return originalPlayer.getName();
	}

	public Status getStatus() {
		return originalPlayer.getStatus();
	}

	public boolean isHumanPlayer() {
		return originalPlayer.isHumanPlayer();
	}

	protected static List<OtherPlayer> convert(List<Player> otherPlayers) {
		return otherPlayers.stream()
				.map(p -> new OtherPlayer(p))
				.collect(Collectors.toList());
	}

}

File: src/main/kotlin/de/spries/fleetcommander/model/facade/OtherPlayer.kt
package de.spries.fleetcommander.model.facade

import de.spries.fleetcommander.model.core.Player

open class OtherPlayer(protected var originalPlayer: Player) {

    fun getName() = originalPlayer.getName()

    fun getStatus() = originalPlayer.getStatus()

    fun isHumanPlayer() = originalPlayer.isHumanPlayer()
}
